File: app/src/main/java/workshop1024/com/xproject/model/news/source/local/NewsDatabase.java
package workshop1024.com.xproject.model.news.source.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.TypeConverters;
import android.content.Context;

import workshop1024.com.xproject.model.ListConverters;
import workshop1024.com.xproject.model.news.News;

@Database(entities = {News.class}, version = 1, exportSchema = false)
@TypeConverters(ListConverters.class)
public abstract class NewsDatabase extends RoomDatabase {
    private static NewsDatabase INSTANCE;

    public abstract NewsDao newsDao();

    private static final Object sLock = new Object();

    public static NewsDatabase getInstance(Context context) {
        synchronized (sLock) {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(context.getApplicationContext(), NewsDatabase.class,
                        "News.db").build();
            }
            return INSTANCE;
        }
    }
}

File: news/src/main/java/workshop1024/com/xproject/news/model/news/local/NewsDatabase.kt
package workshop1024.com.xproject.news.model.news.local

import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

import workshop1024.com.xproject.base.room.converts.XConverters
import workshop1024.com.xproject.news.model.news.News


@Database(entities = [News::class], version = 1, exportSchema = false)
@TypeConverters(XConverters::class)
abstract class NewsDatabase : RoomDatabase() {
    abstract fun newsDao(): NewsDao

    companion object {
        private lateinit var INSTANCE: NewsDatabase

        private val sLock = Any()

        fun getInstance(context: Context): NewsDatabase {
            synchronized(sLock) {
                if (!this::INSTANCE.isInitialized) {
                    INSTANCE = Room.databaseBuilder(context.applicationContext, NewsDatabase::class.java,
                            "newses.db").build()
                }
                return INSTANCE
            }
        }
    }
}